### DISCUSSION TITLE
How to remote start engine

### DISCUSSION THREAD

**Discussion by fromonesource** [Hyundai-Kia-Connect/kia_uvo#452] [Q&A] [answered]
Is it possible to use this to start the engine? I couldn't find anything about it in this repo or [the API](https://github.com/Hyundai-Kia-Connect/hyundai_kia_connect_api).

**Comment by cdnninja** [comment, marked as answer]
Yes.  Start_climate as a service call.

**Reply by esekMy22** [reply]
Is this feature region dependent? I am interested in engine remote start via any app but in EU this option is somehow blocked by KIA - same car model with for USA supports remote engine start.. I can remotely start the engine from outside using the car key. Any idea?

**Reply by cdnninja** [reply]
It's region dependent and only does what the native app can do.

**Reply by esekMy22** [reply]
Thanks, I will do some digging but maybe you can save some of my time and tell me how KIA achieved this that i.e. Remote engine start is supported by their official app in US and not in EU? Would their restrict access to specific API calls based on the car VIN, would they use different API server for US / EU and restrict access to the server or something else? I was going to try that Start climate call for my EU 2022 Kia Stinger which is not electric car but does support  app engine remote start in the US. I assumed Kia did not put much effort on restricting this in the API.. But they rather not implemented this specific API call in the EU version of the app for non-electric cars.

**Reply by cdnninja** [reply]
Each region is a totally separate API.  We have reverse engineered each one.  I would assume this is in the cloud side or head unit as the start command looks very similar.

**Reply by esekMy22** [reply]
Now I understand, thank you, this is much appreciated.

**Reply by rage2021** [reply]
I also have the same probem - i dont see a start engine entity/action..only door unlock.

**Reply by cdnninja** [reply]
> I also have the same probem - i dont see a start engine entity/action..only door unlock.

It isn't an entity but a service call.  Go into developer and services.

**Reply by rage2021** [reply]
thanks this helps.  any idea how to add them to dashboard. i can see i can call the service but not how to add to my dash

**Reply by rage2021** [reply]
can we add service calls to the dashboard? Getting stuck here. whats the difference between entity like lock door vs service call 'remote start engine'. they seem the same to me.

**Comment by fromonesource** [comment]
Oh I thought that was only for HVAC. Thanks! So if I can start engine from the Kia Access app, should I be able to call `kia_uvo.start_climate...` in a HASS automation script and be able to start the car that way too?

**Reply by cdnninja** [reply]
Yes exactly.   I have an Ice car and do this.  I also created a virtual switch that when turns on it fires the service call with my settings and turns the switch off.  This lets google start my car with "hey google turn in the Kia"

**Comment by fromonesource** [comment]
This works great! Thanks!
